lib/IDE/Formatting.cpp

```diff
@@ -2086,6 +2083,32 @@ class FormatWalker : public ASTWalker {
     return Bounds;
   }
 
+  Optional<IndentContext>
+  getIndentContextFromCaseItems(CaseStmt *CS, SourceLoc CloseLoc) {
+    SourceLoc IntroducerLoc = CS->getLoc();
+    if (!isTargetContext(IntroducerLoc, CloseLoc))
+      return None;
+
+    ListAligner Aligner(SM, TargetLocation, IntroducerLoc, IntroducerLoc,
+                        CloseLoc);
+    for (auto &Elem: CS->getCaseLabelItems()) {
+      // Skip the implicit 'error' pattern for catch CaseStmts.
+      if (!Elem.getPattern() || Elem.getPattern()->isImplicit())
+        continue;
```

Comment:
Will this indent something like the following correctly?

```
do {
  throw MyErr.a
} catch where foo == 0,
        where bar == 1 {
}
```

I don't think anyone will be writing code like this in practice, but it is allowed and may be worth testing

Reply:
Good catch! I'll fix it and it to the test cases.